core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQClientFactory.scala

```diff
@@ -263,34 +265,53 @@ private[rabbitmq] class DefaultRabbitMQClientFactory[F[_]: ConcurrentEffect: Tim
 
   private def prepareProducer[A: ProductConverter](producerConfig: ProducerConfig,
                                                    connection: RabbitMQConnection[F],
-                                                   monitor: Monitor[F]): Resource[F, DefaultRabbitMQProducer[F, A]] = {
-    val logger = ImplicitContextLogger.createLogger[F, DefaultRabbitMQProducer[F, A]]
+                                                   monitor: Monitor[F]): Resource[F, BaseRabbitMQProducer[F, A]] = {
+    val logger = ImplicitContextLogger.createLogger[F, BaseRabbitMQProducer[F, A]]
```

Comment:
Don't you want to split it and use according class after you "split the branches" below?